### PR TITLE
bugfix: different company should have different cache key

### PR DESCRIPTION
**Pull request by xiaoronglv**
## Problem

In order to improve the performance, the Workstream cached the access token of each company to the cache. But SDK has been using a static cache key for all companies.

That caused the API call to fail randomly.

## What has been done

1. generate different cache keys for different companies.
2. add test cases.

### CODE REVIEW

File: lib/paylocity_web_service/authentication.rb [hunk 1]
```diff
@@ -26,15 +27,24 @@ def refresh_token
...
       token, expires_in = body['access_token'], body['expires_in']
-      PaylocityWebService::Cache.write(access_token_cache_key, token, expires_in)
+      cache_store.write(access_token_cache_key, token, expires_in: expires_in)
       token
     end
 
     def access_token_cache_key
-      'access_token_cache_key'
+      raise 'company_id is required' if company_id.nil?
+      "PaylocityCompany/#{company_id}/AccessToken/#{endpoint}"
```
Comment (xiaoronglv) [inline lib/paylocity_web_service/authentication.rb:38]: Here is the key line.

File: lib/paylocity_web_service/authentication.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 require 'json'
 require 'base64'
+require 'digest/md5'
```
Comment (rabbit-run) [inline lib/paylocity_web_service/authentication.rb:3]: It seems this is never used. do we need to use it somewhere?

**Review by rabbit-run — APPROVED** [review]
LG other than the inline comment.

**Review by akiratee — APPROVED** [review]

File: lib/paylocity_web_service/authentication.rb (hunk 2 above)
Comment (xiaoronglv) [reply lib/paylocity_web_service/authentication.rb:3]: nice catch!

I have removed this line based on your review.

https://github.com/helloworld1812/paylocity-ruby-gem/commit/0430ce3cb59bb1834552da7cd5775f31c9365f64

**Pull request merged**